include/i2c.h

```diff
@@ -110,16 +117,66 @@ union __deprecated dev_config {
  * These are for internal use only, so skip these in
  * public documentation.
  */
+struct i2c_slave_config;
+
+typedef int (*i2c_slave_write_requested_cb_t)(
+		struct i2c_slave_config *config);
+typedef int (*i2c_slave_read_requested_cb_t)(
+		struct i2c_slave_config *config, u8_t *val);
+typedef int (*i2c_slave_write_received_cb_t)(
+		struct i2c_slave_config *config, u8_t val);
+typedef int (*i2c_slave_read_processed_cb_t)(
+		struct i2c_slave_config *config, u8_t *val);
+typedef int (*i2c_slave_stop_cb_t)(struct i2c_slave_config *config);
+
+struct i2c_slave_callbacks {
+	/** callback function being called when write is requested */
+	i2c_slave_write_requested_cb_t write_requested;
+	/** callback function being called when read is requested */
+	i2c_slave_read_requested_cb_t read_requested;
+	/** callback function being called when byte has been received */
+	i2c_slave_write_received_cb_t write_received;
+	/** callback function being called when byte has been sent */
+	i2c_slave_read_processed_cb_t read_processed;
+	/** callback function being called when stop occurs on the bus */
+	i2c_slave_stop_cb_t stop;
+};
+
+struct i2c_slave_config {
+	/** Private, do not modify */
+	sys_snode_t node;
+	/** Flags for the slave device defined by I2C_SLAVE_FLAGS_* constants */
+	u8_t flags;
+	/** Address for this slave device */
+	u16_t address;
+	/** Callback functions */
+	const struct i2c_slave_callbacks *callbacks;
```

Comment:
move this before flags. you could add also a u8_t _unused; for the last unused byte aligning the struct